frontend/src/components/projectDetail/questionsAndComments.js

```diff
@@ -66,19 +70,23 @@ export const QuestionsAndComments = ({ projectId }) => {
   const token = useSelector((state) => state.auth.get('token'));
   const [comments, setComments] = useState(null);
   const [page, setPage] = useState(1);
+  const isMountedRef = useIsMountedRef();
 
   const handlePagination = (val) => {
     setPage(val);
   };
 
   useEffect(() => {
     if (projectId && page) {
-      fetchLocalJSONAPI(
-        `projects/${projectId}/comments/?perPage=5&page=${page}`,
-        token,
-      ).then((res) => setComments(res));
+      fetchLocalJSONAPI(`projects/${projectId}/comments/?perPage=5&page=${page}`, token).then(
+        (res) => {
+          if (isMountedRef.current) {
+            setComments(res);
+          }
+        },
+      );
     }
-  }, [page, projectId, token]);
+  }, [page, projectId, token, isMountedRef]);
```

Comment:
on that case, wouldn't be better to trigger the request only when `isMountedRef` is `true`? I didn't test on my browser, but maybe it would avoid duplicated requests

Reply:
I'll test it out and see.